docs/README.md

```diff
@@ -0,0 +1,407 @@
+place these files and reference them by URL. We recommend putting them in the
+following dataset:
+[huggingface/documentation-images](https://huggingface.co/datasets/huggingface/documentation-images).
+If an external contribution, feel free to add the images to your PR and ask a
+Hugging Face member to migrate your images to this dataset.
+
+## Styling the docstring
+
+We have an automatic script running with the `make style` comment that will make
+sure that:
+- the docstrings fully take advantage of the line width
+- all code examples are formatted using black, like the code of the 🤗 Optimum
+  library
+
+This script may have some weird failures if you made a syntax mistake or if you
+uncover a bug. Therefore, it's recommended to commit your changes before running
+`make style`, so you can revert the changes done by that script easily.
+
+## Testing documentation examples
+
+Good documentation oftens comes with an example of how a specific function or
+class should be used. Each model class should contain at least one example
+showcasing how to use this model class in inference. *E.g.* the class
+[Wav2Vec2ForCTC](https://huggingface.co/docs/transformers/model_doc/wav2vec2#transformers.Wav2Vec2ForCTC)
+includes an example of how to transcribe speech to text in the [docstring of its
+forward
+function](https://huggingface.co/docs/transformers/model_doc/wav2vec2#transformers.Wav2Vec2ForCTC.forward).
+
+## Writing documenation examples
+
+The syntax for Example docstrings can look as follows:
+
+```
+    Example:
+
+    ```python
+    >>> from transformers import Wav2Vec2Processor, Wav2Vec2ForCTC
+    >>> from datasets import load_dataset
+    >>> import torch
+
+    >>> dataset = load_dataset("hf-internal-testing/librispeech_asr_demo", "clean", split="validation")
+    >>> dataset = dataset.sort("id")
+    >>> sampling_rate = dataset.features["audio"].sampling_rate
+
+    >>> processor = Wav2Vec2Processor.from_pretrained("facebook/wav2vec2-base-960h")
+    >>> model = Wav2Vec2ForCTC.from_pretrained("facebook/wav2vec2-base-960h")
+
+    >>> # audio file is decoded on the fly
+    >>> inputs = processor(dataset[0]["audio"]["array"], sampling_rate=sampling_rate, return_tensors="pt")
+    >>> with torch.no_grad():
+    ...     logits = model(**inputs).logits
+    >>> predicted_ids = torch.argmax(logits, dim=-1)
+
+    >>> # transcribe speech
+    >>> transcription = processor.batch_decode(predicted_ids)
+    >>> transcription[0]
+    'MISTER QUILTER IS THE APOSTLE OF THE MIDDLE CLASSES AND WE ARE GLAD TO WELCOME HIS GOSPEL'
+    ```
+```
+
+The docstring should give a minimal, clear example of how the respective model 
+is to be used in inference and also include the expected (ideally sensible)
+output.
+Often, readers will try out the example before even going through the function 
+or class definitions. Therefore it is of utmost importance that the example 
+works as expected.
+
+# Adding documentation support for an 🤗 Optimum subpackage
```

Comment:
Here's some details on how to add docs for `optimum-intel` and `optimum-graphcore`